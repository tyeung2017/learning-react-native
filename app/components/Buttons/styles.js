import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
  container: {
    alignItems: 'center',
  },
  wrapper: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  icon: {
    width: 19,
    marginRight: 11,
  },
  text: {
    color: '$white',
    fontSize: 18,
    fontWeight: '300',
    paddingVertical: 20,
  },
});

