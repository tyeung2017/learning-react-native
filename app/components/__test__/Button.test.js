import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { ClearButton, styles } from '../Buttons';
import buildStyles from '../../config/styles';

beforeAll(() => buildStyles());

it('it rendererd successfully', () => {
  const rendered = renderer.create(<ClearButton />).toJSON();
  expect(rendered).toBeTruthy();
});

it('styles is an object', () => {
  expect(typeof styles).toBe('object');
});

it('snapshot test for ClearButton', () => {
  const rendered = renderer.create(<ClearButton text="hi there" />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('test button press', () => {
  const callback = sinon.spy();
  const wrapper = shallow(<ClearButton text="button press test" onPress={callback} />);

  expect(wrapper.find(TouchableOpacity).length).toBe(1);

  wrapper
    .find(TouchableOpacity)
    .first()
    .props()
    .onPress();

  expect(callback.called).toBe(true);
});
