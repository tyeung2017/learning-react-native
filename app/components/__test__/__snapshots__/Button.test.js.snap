// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for ClearButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "marginRight": 11,
          "width": 19,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "300",
          "paddingVertical": 20,
        }
      }
    >
      hi there
    </Text>
  </View>
</View>
`;
